### PR TITLE
Implement Frontier integration for EVM logs in pallet-ethy and update…

### PR DESCRIPTION
**Pull request by leo-plugdefi**
# Description
- Implement Frontier-compatible EVM logs integration in `pallet-ethy` and update tests and runtime wiring. Also update fee-proxy runner/impls and add an e2e test that validates FeeProxy log behavior.

## Context & Background
- The project is using Frontier-style EVM log/event plumbing for compatibility with EVM tooling and indexers. This commit wires pallet-ethy into that plumbing so EVM logs emitted by the runtime are compatible with Frontier expectations. Tests were updated and an end-to-end test was added to verify the behavior.

## Related Issues
- https://futureverse.atlassian.net/browse/TRN-1041

Files changed (summary)
- `Cargo.lock` — updated lockfile reflecting dependency changes.
- `Cargo.toml` — top-level manifest changes (workspace dependencies/versions).
- `e2e/test/FeeProxyLogs.test.ts` — NEW: end-to-end test for FeeProxy EVM log behavior.
- `pallet/ethy/Cargo.toml` — pallet manifest updated for Frontier integration changes.
- `pallet/ethy/src/lib.rs` — implement Frontier-compatible EVM log integration (runtime/pallet code).
- `pallet/ethy/src/mock.rs` — mock updates for updated behavior in tests.
- `pallet/fee-proxy/Cargo.toml` — manifest updates for fee-proxy pallet.
- `pallet/fee-proxy/src/impls.rs` — implementation changes to support updated log/runner flow.
- `pallet/fee-proxy/src/runner.rs` — runner changes to align with Frontier log integration.
- `runtime/Cargo.toml` — runtime manifest updated for the new wiring.
- `runtime/src/lib.rs` — runtime wiring to integrate `pallet-ethy` Frontier log support.

Release Notes

## Key Changes
- Added Frontier-compatible EVM log integration to `pallet-ethy`.
- Updated `pallet/fee-proxy` runner/impls to cooperate with the new log flow.
- Added an e2e test `e2e/test/FeeProxyLogs.test.ts` to validate log emission and indexing behavior.
- Workspace manifests and lockfile updated to match the changes.

## Type of Change
- [x] Runtime Changes
- [ ] Client Changes
- This PR changes runtime/pallet behavior (Frontier EVM log integration) and adds tests; no client-visible API changes were introduced.

#### Added
- Frontier-compatible EVM log integration wiring in `pallet-ethy` so runtime-emitted EVM logs conform to Frontier expectations (used by indexers and tooling).

#### Changed
- `pallet-ethy` behavior and its mock/test harness updated to reflect Frontier integration.
- `pallet/fee-proxy` runner/impls updated to match how logs are handled/forwarded.

#### Removed
- None

#### Added
- None

#### Changed
- None

#### Removed
- None

#### Added
- None

#### Changed
- None

#### Removed
- None

#### Added
- None explicitly named in commit — runtime now emits Frontier-style EVM logs (used by event/indexer tooling).

#### Changed
- Emitted EVM log format/forwarding behavior updated for Frontier compatibility.

#### Removed
- None

#### Added
- None

#### Changed
- None

#### Removed
- None

## Storage Migrations
- None

Requirements coverage / status
- Fill PR description with commit details — Done
- Provide files changed and short purpose — Done
- Classify change types (runtime/client/api/etc.) — Done (runtime change)
- Provide test/run instructions — Done
- Link related issues — None present in commit (left as None)

Next steps
- If you want I can:
  - Attach the full diff for the commit to the PR body.
  - Expand the PR with specific code excerpts (not inventing names) for the key runtime changes.
  - Add issue references or checklist items for reviewers.

Would you like the full diff or a concise reviewer checklist added to the PR body?- No new production dependencies were added beyond runtime/pallet changes.
- The e2e test was added to verify log emission and indexing behavior; ensure Docker/Orbstack is running for the e2e environment if required.

## Related Issues
- None linked in the commit. Add issue references here, e.g. `#123`, if you want them attached to the PR.

Files changed (summary)
- `Cargo.lock` — updated lockfile reflecting dependency changes.
- `Cargo.toml` — top-level manifest changes (workspace dependencies/versions).
- `e2e/test/FeeProxyLogs.test.ts` — NEW: end-to-end test for FeeProxy EVM log behavior.
- `pallet/ethy/Cargo.toml` — pallet manifest updated for Frontier integration changes.
- `pallet/ethy/src/lib.rs` — implement Frontier-compatible EVM log integration (runtime/pallet code).
- `pallet/ethy/src/mock.rs` — mock updates for updated behavior in tests.
- `pallet/fee-proxy/Cargo.toml` — manifest updates for fee-proxy pallet.
- `pallet/fee-proxy/src/impls.rs` — implementation changes to support updated log/runner flow.
- `pallet/fee-proxy/src/runner.rs` — runner changes to align with Frontier log integration.
- `runtime/Cargo.toml` — runtime manifest updated for the new wiring.
- `runtime/src/lib.rs` — runtime wiring to integrate `pallet-ethy` Frontier log support.

Release Notes

## Key Changes
- Added Frontier-compatible EVM log integration to `pallet-ethy`.
- Updated `pallet/fee-proxy` runner/impls to cooperate with the new log flow.
- Added an e2e test `e2e/test/FeeProxyLogs.test.ts` to validate log emission and indexing behavior.
- Workspace manifests and lockfile updated to match the changes.

## Type of Change
- [x] Runtime Changes
- [ ] Client Changes
- This PR changes runtime/pallet behavior (Frontier EVM log integration) and adds tests; no client-visible API changes were introduced.

#### Added
- Frontier-compatible EVM log integration wiring in `pallet-ethy` so runtime-emitted EVM logs conform to Frontier expectations (used by indexers and tooling).

#### Changed
- `pallet-ethy` behavior and its mock/test harness updated to reflect Frontier integration.
- `pallet/fee-proxy` runner/impls updated to match how logs are handled/forwarded.

#### Removed
- None

#### Added
- None

#### Changed
- None

#### Removed
- None

#### Added
- None

#### Changed
- None

#### Removed
- None

#### Added
- None explicitly named in commit — runtime now emits Frontier-style EVM logs (used by event/indexer tooling).

#### Changed
- Emitted EVM log format/forwarding behavior updated for Frontier compatibility.

#### Removed
- None

#### Added
- None

#### Changed
- None

#### Removed
- None

## Storage Migrations
- None

### CODE REVIEW

**Review by JCSanPedro — CHANGES_REQUESTED** [review]
Nice!! I can confirm that it is working for the test cases that I had 👍 👍 

Though there a couple of edge cases that need some attention.

1. Getting duplicate eth logs. This happens when we invoke the contract through the `eth_sendTransaction` rpc call. Wrote a [test case in this branch](https://github.com/futureversecom/trn-seed/blob/jn/fix-eth-logs/e2e/test/FeeProxyLogs.test.ts#L162)
2. The synthetic transaction hash can't be used to retrieve the transaction through the  eth rpc. Is it possible to create the real tx hash since we have all of the values for the unsigned tx? Test case [here](https://github.com/futureversecom/trn-seed/blob/jn/fix-eth-logs/e2e/test/FeeProxyLogs.test.ts#L177)
3. Because we append the transaction statuses to the existing transaction statuses, the ordering of the logs may be incorrect when two transactions in the same block?

**Comment by leo-plugdefi** [comment]
> Nice!! I can confirm that it is working for the test cases that I had 👍 👍
> 
> Though there a couple of edge cases that need some attention.
> 
> 1. Getting duplicate eth logs. This happens when we invoke the contract through the `eth_sendTransaction` rpc call. Wrote a [test case in this branch](https://github.com/futureversecom/trn-seed/blob/jn/fix-eth-logs/e2e/test/FeeProxyLogs.test.ts#L162)
> 2. The synthetic transaction hash can't be used to retrieve the transaction through the  eth rpc. Is it possible to create the real tx hash since we have all of the values for the unsigned tx? Test case [here](https://github.com/futureversecom/trn-seed/blob/jn/fix-eth-logs/e2e/test/FeeProxyLogs.test.ts#L177)
> 3. Because we append the transaction statuses to the existing transaction statuses, the ordering of the logs may be incorrect when two transactions in the same block? I'm not sure if this can even be avoided, or much of an impact it would have. But it would of course affect any indexers relying on the eth logs.
> 
> Also just wondering why the hooks were added to the `ethy` pallet? The `ethy` pallet is used for bridging, and isn't related to this issue I believe? Would it make sense to add the additional business logic to the hooks in the [ethereum](https://github.com/futureversecom/trn-frontier/blob/polkadot-v0.9.30-TRN/frame/ethereum/src/lib.rs#L169) or [evm](https://github.com/futureversecom/trn-frontier/blob/polkadot-v0.9.30-TRN/frame/evm/src/lib.rs#L100) pallet in the frontier lib?

- Practical reason: ethy is already present in every block, has hooks, and is a convenient place to stage/merge EVM call activity behind a feature gate. We made it explicitly opt-in via the
Config::FrontierLogMerge strategy type and the “frontier-logs” feature.
- That said, you’re right that this isn’t a bridge concern. Two better options:
    - A tiny dedicated pallet (e.g., pallet-evm-logs-canonicalizer) that owns the staging/merge concerns and nothing else. (this is overkill I feel)
    - Upstream Frontier integration (in pallet-ethereum or pallet-evm) so the behavior is first-class and available to all chains. (this is less ideal I feel, need to maintain a fork)

**Comment by JCSanPedro** [comment]
> this is less ideal I feel, need to maintain a fork

Yeah I was talking about updating our fork which we already maintain - https://github.com/futureversecom/trn-frontier

File: e2e/test/FeeProxyLogs.test.ts [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+      fromBlock: blockNumber,
+      toBlock: blockNumber,
+    });
+    // Expect exactly one Transfer log from this call within the block
+    // (deduplication avoids staging the same log twice)
+    const count = logs.filter((l) => l.address.toLowerCase() === token.address.toLowerCase()).length;
+    expect(count).to.eq(1);
+  });
+
+  it("synthetic transaction hashes from extrinsic path are not retrievable via eth_getTransactionByHash", async () => {
```
[the remark this answers is not in the document]
Comment (JCSanPedro) [reply e2e/test/FeeProxyLogs.test.ts:163]: I'm wondering if this behaviour is something we need to document somewhere for other builders

File: e2e/test/FeeProxyLogs.test.ts [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
```
[the remark this answers is not in the document]
Comment (leo-plugdefi) [reply e2e/test/FeeProxyLogs.test.ts:163]: interesting, I thought CI shall picked them up

**Review by JCSanPedro — APPROVED** [review]
